lib/vmdb/permission_stores.rb

```diff
@@ -1,34 +1,33 @@
-module Vmdb
-  module PermissionStores
-    class Configuration
-      attr_accessor :backend
-      attr_accessor :options
+require "yaml"
 
-      def initialize
-        @options = {}
-      end
+module Vmdb
+  class PermissionStores
+    def self.instance
+      @instance ||= new(blacklist)
+    end
 
-      def create
-        PermissionStores.create(self)
-      end
+    def self.blacklist
+      permission_files.flat_map { |file| YAML.load_file(file) }
+    end
 
-      def load
-        require "vmdb/permission_stores/#{backend}"
-      end
+    private_class_method def self.permission_files
+      Vmdb::Plugins.to_a.unshift(Rails)
+        .map { |source| source.root.join("config", "permissions.yml") }
+        .select(&:exist?)
     end
 
-    class << self
-      attr_accessor :configuration, :instance
+    attr_reader :blacklist
+
+    def initialize(blacklist)
+      @blacklist = blacklist
     end
 
-    def self.configure
-      @configuration = Configuration.new
-      yield @configuration
+    def can?(permission)
+      blacklist.exclude?(permission.to_s)
```

Comment:
Seems like this will fail when blacklist is nil

Reply:
blacklist won't be nil as it's initialized with an empty Array when nothing exists.